src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -1,30 +1,154 @@
+
+def list_sps(spn=None, display_name=None, query_filter=None):
+    client = _graph_client_factory()
+
+    sub_filters = []
+    if query_filter:
+        sub_filters.append(query_filter)
+    if spn:
+        sub_filters.append("servicePrincipalNames/any(c:c eq '{}')".format(spn))
+    if display_name:
+        sub_filters.append("startswith(displayName,'{}')".format(display_name))
+
+    return client.service_principals.list(filter=(' and '.join(sub_filters)))
+
+def list_users(upn=None, display_name=None, query_filter=None):
+    client = _graph_client_factory()
+
+    sub_filters = []
+    if query_filter:
+        sub_filters.append(query_filter)
+    if upn:
+        sub_filters.append("userPrincipalName eq '{}'".format(upn))
+    if display_name:
+        sub_filters.append("startswith(displayName,'{}')".format(display_name))
+
+    return client.users.list(filter=(' and ').join(sub_filters))
+
+def create_user(user_principal_name, display_name, password, mail_nick_name=None, #pylint: disable=too-many-arguments
+                immutable_id=None, force_change_password_next_login=False):
+    '''
+    :param mail_nick_name: mail alias. default to user principal name
+    '''
+    client = _graph_client_factory()
+    mail_nick_name = mail_nick_name or user_principal_name.split('@')[0]
+    param = UserCreateParameters(user_principal_name=user_principal_name, account_enabled=True,
+                                 display_name=display_name, mail_nickname=mail_nick_name,
+                                 immutable_id=immutable_id,
+                                 password_profile=PasswordProfile(
+                                     password, force_change_password_next_login))
+    return client.users.create(param)
+
+create_user.__doc__ = UserCreateParameters.__doc__
+
+def list_groups(display_name=None, query_filter=None):
+    client = _graph_client_factory()
+
+    sub_filters = []
+    if query_filter:
+        sub_filters.append(query_filter)
+    if display_name:
+        sub_filters.append("startswith(displayName,'{}')")
+
+    return client.groups.list(filter=(' and ').join(sub_filters))
+
+def _get_object_id(assignee):
+    client = _graph_client_factory()
+    result = None
+    if assignee.find('@') >= 0: #looks like a user principal name
+        result = list(client.users.list(filter="userPrincipalName eq '{}'".format(assignee)))
+    if not result:
+        result = list(client.service_principals.list(
+            filter="servicePrincipalNames/any(c:c eq '{}')".format(assignee)))
+    if not result: #assume an object id, let us verify it
+        from azure.graphrbac.models import GetObjectsParameters
+        result = list(client.objects.get_objects_by_object_ids(
+            GetObjectsParameters(include_directory_object_references=True, object_ids=[assignee])))
+
+    if len(result) > 1:
+        err = "More than one matched objects for '{}'. Please use object id"
```

Comment:
On a second thought, more than one match should never happen, so let me just remove this whole `if`.